Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/Networking/Forge/Networking/FacepunchP2PClient.cs

```diff
@@ -0,0 +1,430 @@
+
+			// By default pending creates should be true and flushed when ready
+			if (!pendCreates)
+				PendCreates = true;
+
+			try
+			{
+				ushort clientPort = DEFAULT_PORT;
+
+				// Make sure not to listen on the same port as the server for local networks
+				if (clientPort == DEFAULT_PORT)
+					clientPort++;
+
+				Client = new CachedFacepunchP2PClient(hostId);
+
+				// Do any generic initialization in result of the successful bind
+				OnBindSuccessful();
+
+				// Get a random hash key that needs to be used for validating that the server was connected to
+				headerHash = Websockets.HeaderHashKey();
+
+				// This is a typical Websockets accept header to be validated
+				byte[] connectHeader = Websockets.ConnectionHeader(headerHash, DEFAULT_PORT);
+
+				// Setup the identity of the server as a player
+				server = new NetworkingPlayer(0, hostId, true, this);
+
+				// Create the thread that will be listening for new data from connected clients and start its execution
+				Task.Queue(ReadNetwork);
+
+				//Let myself know I connected successfully
+				OnPlayerConnected(server);
+
+				// Set myself as a connected client
+				server.Connected = true;
+
+				//Set the port
+				SetPort(clientPort);
+
+				int connectCounter = 0;
+				Task.Queue(() =>
+				{
+					do
+					{
+						// Send the accept headers to the server to validate
+						Client.Send(connectHeader, connectHeader.Length, hostId, P2PSend.Reliable);
+						Thread.Sleep(3000);
+					} while (!headerExchanged && IsBound && ++connectCounter < CONNECT_TRIES);
+
+					if (connectCounter >= CONNECT_TRIES)
+					{
+						if (connectAttemptFailed != null)
+							connectAttemptFailed(this);
+					}
+				});
+			}
+			catch (Exception e)
+			{
+				Logging.BMSLog.LogException(e);
+				// Do any generic initialization in result of the binding failure
+				OnBindFailure();
+
+				throw new FailedBindingException("Failed to bind to server's SteamId, see inner exception", e);
+			}
+		}
+
+		/// <summary>
+		/// Connects to a Steam Lobby then kicks off connection to the lobby owner's SteamId
+		/// </summary>
+		/// <param name="lobbyToJoin">The <see cref="Steamworks.Data.Lobby"/> to join</param>
+		/// <param name="pendCreates">Set the NetWorker::PendCreates to true</param>
+		private async void ConnectToLobbyAsync(Steamworks.Data.Lobby lobbyToJoin, bool pendCreates)
+		{
+			RoomEnter roomEnter = await lobbyToJoin.Join();
+			if (roomEnter != RoomEnter.Success)
+				return;
+			Lobby = lobbyToJoin;
+			Connect(Lobby.Owner.Id, pendCreates);
+		}
+
+		/// <summary>
+		/// Disconnect this client from the server
+		/// </summary>
+		/// <param name="forced">Used to tell if this disconnect was intentional <c>false</c> or caused by an exception <c>true</c></param>
+		public override void Disconnect(bool forced)
+		{
+			Logging.BMSLog.Log("<color=cyan>FacepunchP2P client disconnecting...</color>");
+
+			if (Lobby.Id.Value > 0)
```

Comment:
Can this be negative? If not why not check if it is `!= 0`?

Reply:
No, and no reason, so yes!